### PR TITLE
[PPS] Added endpoint to get all reports of student in JSON

### PR DESCRIPTION
**Pull request by pritamps**
* Added a new secondary index to dynamodb -- our reports db is now queryable by userid as well
* Added a new endpoint: /reports/student_reports/4444444444?format=json
Sample output:

```
{"student_id":"4444444444","reports":[{"test_name":"Test - Homework Quiz","test_session_id":"Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823","rank":"1.0","percentile":80,"report_link":"https://reports.avantifellows.org/reports/student_quiz_report/Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823/4444444444"},{"test_name":"Test - Homework Quiz","test_session_id":"Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823","rank":"1.0","percentile":90,"report_link":"https://reports.avantifellows.org/reports/student_quiz_report/Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823/4444444444"},{"test_name":"Test - Homework Quiz","test_session_id":"Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823","rank":"2.0","percentile":80,"report_link":"https://reports.avantifellows.org/reports/student_quiz_report/Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823/4444444444"},{"test_name":"Test - Homework Quiz","test_session_id":"Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823","rank":"3.0","percentile":40,"report_link":"https://reports.avantifellows.org/reports/student_quiz_report/Test_-_Homework_Quiz_2022-07-27_62e246c6ed7e4cbb6a78b823/4444444444"}]}
``

### CODE REVIEW

File: generate_table/__main__.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-
+import boto3
 import os
 
 # Update and use `.env.prod` to write to prod dynamodb table
 # or use `.env.staging` to write to staging dynamodb table
 # TODO: Staging table doesn't exist as of now :p - so maybe we should create it in the future
 load_dotenv(".env.local")
 
+print("HELLO")
```
Comment (suryabulusu) [inline generate_table/__main__.py:16]: we can comment out these print statements

File: generate_table/__main__.py [hunk 2]
```diff
@@ -35,5 +40,15 @@ def drop_tables():
...
 
 
+def add_secondary_ind():
+    # DON'T RUN THIS for the key user_id key as it's already been run
+    ddb = initialize_db()
+    add_secondary_index(ddb)
+
+
+def drop_secondary_ind():
+    drop_secondary_index("gsi_user_id-section")
```
Comment (suryabulusu) [inline generate_table/__main__.py:50]: shouldn't this be "gsi_user_Id"

File: generate_table/__main__.py (hunk 1 above)
Comment (pritamps) [reply generate_table/__main__.py:16]: Very polite, thank you :D will remove!

File: generate_table/__main__.py (hunk 2 above)
Comment (pritamps) [reply generate_table/__main__.py:50]: So these methods are meant to be run manually, mostly one-time. Will actually make this a parameter for the function.

File: generate_table/__main__.py [hunk 3]
```diff
@@ -35,5 +40,15 @@ def drop_tables():
     drop_student_quiz_reports(ddb)
 
 
+def add_secondary_ind():
+    # DON'T RUN THIS for the key user_id key as it's already been run
```
Comment (suryabulusu) [inline generate_table/__main__.py:44]: but this is getting run by default in `if __name__ == "__main__" ? 🤔

File: generate_table/__main__.py [hunk 4]
```diff
@@ -26,14 +27,36 @@ def initialize_db():
...
+def add_secondary_ind():
+    ddb = initialize_db()
+    add_secondary_index(ddb)
+
+
+def drop_secondary_ind(index_name: str):
+    """
+    Drops a secondary index
+    """
+    drop_secondary_index(index_name)
```
Comment (suryabulusu) [inline generate_table/__main__.py:54]: `drop_secondary_index(ddb, index_name)`

Comment (suryabulusu) [reply generate_table/__main__.py:54]: and `ddb = initialize_db()`

**Review by suryabulusu — APPROVED** [review]

**Pull request merged**